x-pack/plugins/secops/server/lib/network_top_n_flow/elasticsearch_adapter.ts

```diff
@@ -77,3 +50,44 @@ export class ElasticsearchNetworkTopNFlowAdapter implements NetworkTopNFlowAdapt
     };
   }
 }
+
+const getTopNFlowEdges = (
+  response: DatabaseSearchResponse<NetworkTopNFlowData, TermAggregation>,
+  options: NetworkTopNFlowRequestOptions
+): NetworkTopNFlowEdges[] => {
+  if (options.networkTopNFlowDirection === NetworkTopNFlowDirection.uniDirectional) {
+    return formatTopNFlowEgdes(
+      getOr([], 'aggregations.top_uni_flow.buckets', response),
+      options.networkTopNFlowType
+    );
+  }
+  return formatTopNFlowEgdes(
+    getOr([], 'aggregations.top_bi_flow.buckets', response),
+    options.networkTopNFlowType
+  );
+};
+
+const formatTopNFlowEgdes = (
+  buckets: NetworkTopNFlowBuckets[],
+  networkTopNFlowType: NetworkTopNFlowType
+): NetworkTopNFlowEdges[] =>
+  buckets.map((bucket: NetworkTopNFlowBuckets) => ({
+    node: {
+      _id: bucket.key,
+      timestamp: bucket.timestamp.value_as_string,
+      [networkTopNFlowType]: {
+        count: bucket.ip_count.value || null,
+        domain: bucket.domain.buckets.map(bucketDomain => bucketDomain.key) || null,
```

Comment:
I don't think

```ts
domain: bucket.domain.buckets.map(bucketDomain => bucketDomain.key) || null,
```
That can ever result in a `null` can it? You'll always at least get an empty array.